src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -400,15 +402,37 @@ private LanguageExpression ConvertAccessExpression(AccessExpression expression,
             }
         }
 
-        private (LanguageExpression @base, IEnumerable<LanguageExpression> properties, bool safeAccess) ConvertModulePropertyAccess(ModuleReferenceExpression reference, PropertyAccessExpression expression) => expression.PropertyName switch
+        private (LanguageExpression @base, IEnumerable<LanguageExpression> properties, bool safeAccess) ConvertModulePropertyAccess(ModuleReferenceExpression reference, PropertyAccessExpression expression)
         {
-            // the name is dependent on the name expression which could involve locals in case of a resource collection
-            "name" => (GetModuleNameExpression(reference.Module), Enumerable.Empty<LanguageExpression>(), false),
-            "outputs" => (GetModuleReferenceExpression(reference.Module, reference.IndexContext, false),
+            switch (expression.PropertyName)
+            {
+
+                case "name":
+                    // the name is dependent on the name expression which could involve locals in case of a resource collection
+
+                    return (GetModuleNameExpression(reference.Module), Enumerable.Empty<LanguageExpression>(), false);
+
+                case "outputs":
+                    var moduleSymbol = reference.Module;
+                    var bodyObjectType = moduleSymbol.TryGetBodyObjectType();
+                    var hasSecureOutput = bodyObjectType?.Properties.Any(p => p.Value.Flags.HasFlag(TypeSystem.Types.TypePropertyFlags.WriteOnly)) ?? false;
```

Comment:
@jeskew do we have any existing methods we can use to check recursively whether a type is secure?

Reply:
https://github.com/Azure/bicep/blob/main/src/Bicep.Core/Analyzers/Linter/Common/FindPossibleSecretsVisitor.cs will do it. That was written to match the redaction logic in ARM.